### ISSUE TITLE
Crashes after losing connection to ATEM

### ISSUE THREAD

**Issue by f-fritz** [danielbuechele/atemOSC#113]
I'm experiencing a very annoying behavior as the App crashes or quits a few seconds after the connection to the ATEM switcher was lost when a connection existed before.

Starting the App with the ATEM being unavailable, atemOSC shows the red square and behaves as expected.
Starting the App with the ATEM available, atemOSC shows the green square. When the connection is interrupted in this state and the ATEM unavailable for I guess >5sec, atemOSC crashes or quits. I've seen both but am not quite sure what happens when.
We had some trouble with our network with the ATEM being unavailable very frequently and up to 2mins so atemOSC kept crashing every few minutes.

Tried every version of atemOSC since 2.5.2 - worked very well with 2.5.3 for the last months

I'm not sure if this is related to the problem above, but I took a look at Console>Diagnostic Reports and there were some logs of macOS terminating atemOSC due to ~850 activations/second (max 150 / second allowed). I don't have the logs available at the moment, but I can try to get and upload them.

**Comment by SteffeyDev** [comment]
I'm working on 2.5.4 which may fix this... I found some code in the function called when the Item disconnects that may have crashed in a race condition (ae0044efd7c9bd4c9e4d3cc4c751c9b3d958e008)

**Comment by SteffeyDev** [comment]
I'm assuming this is similar to #103, which seemed to be related to network issues and thus disconnects.

**Comment by SteffeyDev** [comment]
The full crash report would still help, though

**Comment by SteffeyDev** [comment]
@f-fritz If you are still experiencing this issue, can you try the latest version of atemOSC and see if you still experience the problem?

**Comment by f-fritz** [comment]
Back then, I had a small chain of tools (including atemOS) converting protocols. This worked fine at the beginning but then got increasingly unreliable and hadn't been very user friendly at starting up the other tools. As atemOSC began to crash in December 2018 and I couldn't figure out whether it was my fault or a bug, I wrote my own tool, completely skipping the OSC part and reducing my chain of converters to one single conversion. Hence, there was no more need to use atemOSC. Moreover, I barely had time and chances to (thoroughly) test newer versions of atemOSC - and as my tool did the job, I didn't need to. I didn't even come about getting the logs, mentioned in my report. (And I won't be able to in the near future as I am currently no longer using the ATEM for that production.)

But as no other affected people reacted on this report I guess this had either been a configuration error or has already been fixed.
